include/mata/nfa/nfa.hh

```diff
@@ -562,15 +562,37 @@ OnTheFlyAlphabet create_alphabet(const std::vector<const Nfa*>& nfas);
 Nfa union_nondet(const Nfa &lhs, const Nfa &rhs);
 
 /**
- * @brief Compute union by product construction.
+ * @brief Compute union of two complete deterministic NFAs. Perserves determinism.
  *
- * Preserves determinism.
- * @param[in] first_epsilon The first symbol to handle as an epsilon.
- * @param[out] prod_map Map mapping product states to the original states.
- * @return Union by product construction of @p lhs and @p rhs.
+ * The union is computed by product construction with or condition on the final states.
+ * @param lhs First complete deterministic automaton.
+ * @param rhs Second complete deterministic automaton.
  */
-Nfa union_product(const Nfa &lhs, const Nfa &rhs, Symbol first_epsilon = EPSILON,
-                  std::unordered_map<std::pair<State,State>,State> *prod_map = nullptr);
+Nfa union_det_complete_by_product_or(const Nfa &lhs, const Nfa &rhs);
```

Comment:
I suggest only `union_det_complete()`. The variant, if there are more variants in the future, will be selected through opt params.